src/apps/yield-protocol/arbitrum/yield-protocol.balance-fetcher.ts

```diff
@@ -0,0 +1,209 @@
+  ) {}
+
+  private async getBorrowBalances(address: string) {
+    const data = await this.appToolkit.helpers.theGraphHelper.request<YieldVaultRes>({
+      endpoint: yieldV2ArbitrumSubgraph,
+      query: vaultsQuery,
+      variables: { address },
+    });
+    const vaultOwner = data.vaultOwner;
+
+    if (!vaultOwner) return [];
+
+    const positions = await Promise.all(
+      vaultOwner.vaults.map(async vault => {
+        const {
+          debtAmount,
+          collateralAmount,
+          series: {
+            baseAsset: { id: artAddress, assetId: baseId },
+            fyToken: { maturity },
+          },
+          collateral: {
+            asset: { id: ilkAddress, assetId: ilkId },
+          },
+        } = vault;
+
+        // get the corresponding art (debt) and ilk (collateral) of the vault
+        const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
+        const art = baseTokens.find(v => v.address === artAddress.toLowerCase());
+        const ilk = baseTokens.find(v => v.address === ilkAddress.toLowerCase());
+
+        if (!art || !ilk) return null;
+
+        // data props
+        const collateral = drillBalance(ilk, parseUnits(collateralAmount.toString(), ilk.decimals).toString());
+        const debt = drillBalance(art, parseUnits(debtAmount.toString(), art.decimals).toString(), { isDebt: true });
+        const tokens = [collateral, debt];
+        const balanceUSD = sumBy(tokens, v => v.balanceUSD);
+        const collateralizationRatio = `${(debt.balanceUSD === 0
+          ? 0
+          : (collateral.balanceUSD / Math.abs(debt.balanceUSD)) * 100
+        )
+          .toFixed(2)
+          .toString()}%`;
+
+        const multicall = this.appToolkit.getMulticall(network);
+        const cauldron = this.yieldProtocolContractFactory.cauldron({ address: CAULDRON, network });
+        const [{ dec: cauldronPairDecimals }, { ratio }] = await Promise.all([
+          multicall.wrap(cauldron).debt(baseId, ilkId),
+          multicall.wrap(cauldron).spotOracles(baseId, ilkId),
+        ]);
+        const minCollatRatio = ratio / 10 ** cauldronPairDecimals;
+        const minCollateralizationRatio = `${(minCollatRatio * 100).toString()}%`;
+        const liqPrice = (debtAmount * minCollatRatio) / collateralAmount;
+        const liquidationPrice = `1 ${ilk.symbol}: ${liqPrice.toFixed(2)} ${art.symbol}`;
+
+        // display props
+        const displayProps: DisplayProps = {
+          label: `Yield Vault`,
+          secondaryLabel: `${getLabelFromToken(art)} Debt and ${getLabelFromToken(ilk)} Collateral`,
+          tertiaryLabel: formatMaturity(maturity),
+          images: [getImagesFromToken(art)[0], getImagesFromToken(ilk)[0]],
+        };
+
+        const positionBalance: ContractPositionBalance<YieldVaultContractPositionDataProps> = {
+          type: ContractType.POSITION,
+          address,
```

Comment:
I think you want this address to be `0x16e25cf364cecc305590128335b8f327975d0560`. If I understand correctly, a user interacts with this address to open their vault, i.e.: [here](https://arbiscan.io/address/0x16e25cf364cecc305590128335b8f327975d0560?fromaddress=0x35e057137060cd397bd82e3dff54beba480e7012) is a list of transactions for the example address with vault balances you gave in your description.

Reply:
cool! that's the `ladle` contract, which interacts with the `cauldron` contract, where the `cauldron` is the contract that holds the vaults. guessing i should use the cauldron then?

Reply:
As mentioned below, it'd be better to use the `ladle` address because it'd allow the Zapper backend to intelligently determine if a user has Yield Protocol vault open by checking if the user has a transaction to this address (the user interacts with the `ladle`, not the `cauldron`).

Reply:
perfect! this has been updated.